sympy/stats/joint_rv_types.py

```diff
@@ -394,6 +394,100 @@ def MultivariateEwens(syms, n, theta):
     """
     return multivariate_rv(MultivariateEwensDistribution, syms, n, theta)
 
+#-------------------------------------------------------------------------------
+# Generalized Multivariate Log Gamma distribution ---------------------------------------------------------
+
+class GeneralizedMultivariateLogGammaDistribution(JointDistribution):
+
+    _argnames = ['omega', 'v', 'l', 'mu']
+    is_Continuous=True
+
+    def check(self, omega, v, l, mu):
+        _value_check(omega.is_square, "the matrix must be square")
+        for val in omega.values():
+            _value_check((val >= 0) != False and (val <= 1) != False,
+                "all values in matrix must be between 0 and 1(both inclusive).")
+        _value_check(omega.diagonal().equals(ones(1, omega.shape[0])),
+                        "all the elements of diagonal should be 1.")
+        _value_check((v > 0) != False, "v must be positive")
+        for lk in l:
+            _value_check((lk > 0) != False, "lamda must be a positive vector.")
+        for muk in mu:
+            _value_check((muk > 0) != False, "mu must be a positive vector.")
+        _value_check(omega.shape == (len(l), len(mu)),
+                        "lamda, mu should be of same length and omega should "
+                        " be of shape (length of lamda, length of mu)")
+
+    @property
+    def set(self):
+        from sympy.sets.sets import Interval
+        return S.Reals**len(self.l)
+
+    def pdf(self, *y):
+        from sympy.functions.special.gamma_functions import gamma
+        omega, v, l, mu = self.omega, self.v, self.l, self.mu
+        n = Symbol('n', negative=False, integer=True)
+        k = len(l)
+        d = Pow(Rational(omega.det()), Rational(1, k - 1))
```

Comment:
More simply, `d = omega.det()**Rational(1, k - 1)`. Can `k` be `0` or `1`?

Reply:
I think we should keep k > 1 since, k = 1 doesn't any practical sense, `omega` has correlation coefficient as its element which will make sense only when at least two RVs will be there. I will update it today. Thanks for pointing out.

Reply:
Did you address this?

Reply:
I will add the test and since test will run again I will simplify the other test too.